pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -112,34 +102,17 @@ func resourceArtifactoryLdapSetting() *schema.Resource {
 			Description: `(Optional) When set, supports paging results for the LDAP server. This feature requires that the LDAP server supports a PagedResultsControl configuration. Default value is "true".`,
 		},
 		"search_filter": {
-			Type:     schema.TypeString,
-			Optional: true,
-			ValidateDiagFunc: validation.ToDiagFunc(validation.All(
-				validation.StringIsNotEmpty,
-				func(value interface{}, key string) ([]string, []error) {
-					_, err := ldap.CompileFilter(value.(string))
-					if err != nil {
-						return nil, []error{err}
-					}
-					return nil, nil
-				},
-			)),
-			Description: "(Optional) A filter expression used to search for the user DN used in LDAP authentication. This is an LDAP search filter (as defined in 'RFC 2254') with optional arguments. In this case, the username is the only argument, and is denoted by '{0}'. Possible examples are: (uid={0}) - This searches for a username match on the attribute. Authentication to LDAP is performed from the DN found if successful.",
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringIsNotEmpty, validateLdapFilter)),
+			Description:      "(Optional) A filter expression used to search for the user DN used in LDAP authentication. This is an LDAP search filter (as defined in 'RFC 2254') with optional arguments. In this case, the username is the only argument, and is denoted by '{0}'. Possible examples are: (uid={0}) - This searches for a username match on the attribute. Authentication to LDAP is performed from the DN found if successful.",
 		},
 		"search_base": {
-			Type:     schema.TypeString,
-			Optional: true,
-			Default:  "",
-			ValidateDiagFunc: validation.ToDiagFunc(
-				func(value interface{}, key string) ([]string, []error) {
-					_, err := ldap.ParseDN(value.(string))
-					if err != nil {
-						return nil, []error{err}
-					}
-					return nil, nil
-				},
-			),
-			Description: "(Optional) A context name to search in relative to the base DN of the LDAP URL. For example, 'ou=users' With the LDAP Group Add-on enabled, it is possible to enter multiple search base entries separated by a pipe ('|') character.",
+			Type:             schema.TypeString,
+			Optional:         true,
+			Default:          "",
+			ValidateDiagFunc: validation.ToDiagFunc(validateLdapDn),
```

Comment:
Your validation is inconsistent. In someplaces you do:
`validation.ToDiagFunc(validation.All(validation.StringIsNotEmpty, validateLdapDn))`
and here you don't.

Reply:
See below. Here this should just be:
`ValidateDiagFunc: validateLdapDn`